### PR TITLE
docs: add common errors section for update-ca-certificates when rehashing fails

### PR DESCRIPTION
**Pull request by reubenmiller**
## Proposed changes

Document a common errors with the `update-ca-certificates` when the device has an older version of `openssl` which does not support the `rehash` command resulting in mosquitto being not able to connect to the cloud due to ssl verify errors.

The documentation also includes a work around to manually generate the ca-certificate hashes/symlinks.

This problem was first observed on Opto22 devices.

## Types of changes

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Improvement (general improvements like code refactoring that doesn't explicitly fix a bug or add any new functionality)
- [x] Documentation Update (if none of the other choices apply)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Paste Link to the issue
<br/>

## Checklist

- [x] I have read the [CONTRIBUTING](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://github.com/thin-edge/thin-edge.io/blob/main/CONTRIBUTOR-LICENSE-AGREEMENT.md) (in all commits with git commit -s)
- [ ] I ran `cargo fmt` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I used `cargo clippy` as mentioned in [CODING_GUIDELINES](https://github.com/thin-edge/thin-edge.io/blob/main/CODING_GUIDELINES.md)
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: docs/src/operate/security/cloud-authentication.md [hunk 1]
```diff
@@ -90,3 +98,77 @@ Additionally, you can check correctness of the installed certificate:
...
+
+#### update-ca-certificates fails with `'rehash' is an invalid command` {#update-ca-certificates-rehash-error}
+
+Older **openssl** versions (e.g. `OpenSSL 1.0.2d 9 Jul 2015`), don't support the `rehash` command, which will result in the `update-ca-certificates` command showing the following error:
+
+```sh
+Updating certificates in /etc/ssl/certs...
+openssl:Error: 'rehash' is an invalid command.
+
+Standard commands
```
[the remark this answers is not in the document]
Comment (reubenmiller) [reply docs/src/operate/security/cloud-authentication.md:112]: The reasoning behind was that given that openssl spits out this large output, that users might overlook the first line...but yeah it does seem a bit noisy for the docs.

File: docs/src/operate/security/cloud-authentication.md [hunk 2]
```diff
@@ -90,3 +98,77 @@ Additionally, you can check correctness of the installed certificate:
...
+idea-cbc          idea-cfb          idea-ecb          idea-ofb          
+rc2               rc2-40-cbc        rc2-64-cbc        rc2-cbc           
+rc2-cfb           rc2-ecb           rc2-ofb           rc4               
+rc4-40            seed              seed-cbc          seed-cfb          
+seed-ecb          seed-ofb
+```
+
+By default, mosquitto requires each ca-certificate to be hashed and without it mosquitto will not be able to connect to the cloud due to SSL verify errors (visible from the mosquitto logs).
+
+The problem can be fixed by running the following one-liner which generates the ca-certificate hashes manually. Beware you, will have to run the one-liner each time you add/remove a ca-certificate using `update-ca-certificates`:
```
[the remark this answers is not in the document]
Comment (reubenmiller) [reply docs/src/operate/security/cloud-authentication.md:150]: Let me see if I can reword it to be more clear...and fix the grammer of `Beware`

File: docs/src/operate/security/cloud-authentication.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (reubenmiller) [reply docs/src/operate/security/cloud-authentication.md]: Resolved by e68965eb7f056194a56c7ecddd18d7f91d18d8ca

File: docs/src/operate/security/cloud-authentication.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (reubenmiller) [reply docs/src/operate/security/cloud-authentication.md]: Resolved by e68965eb7f056194a56c7ecddd18d7f91d18d8ca

**Review by didier-wenzek — APPROVED** [review]
Approved
